### ISSUE TITLE
toll information in route results

### ISSUE THREAD

**Issue by linuzer** [Project-OSRM/osrm-backend#4901]
Hi,

I need to calculate truck-toll information on routes. I’m not sure if this information is natively available in OSM data — that would be the best case. But alternatively I could calculate the toll by myself afterwards, if OSRM could give me back the type of street for each section of the route (or as a summary per street-type). As far as I checked the documentation, I was not able to find that type of information in the result.

I’m aware of [#1472](https://github.com/Project-OSRM/osrm-backend/issues/1472) and [#1342](https://github.com/Project-OSRM/osrm-backend/issues/1342) and related issues and this would be a possibility to get the street-type.

But is there maybe a better way to do it? Are there anywhere already truck-toll-related thoughts in the OSRM-project?

Thanks a lot!

Edit: Clarified, that I need truck-toll

**Comment by TheMarex** [comment]
We have the [`classes`](https://github.com/Project-OSRM/osrm-backend/blob/master/docs/http.md#intersection-object) property that should contain a `toll` value if the OSM way is tagged with `toll = yes`. Hope that helps.

**Comment by linuzer** [comment]
Thanks for the hint! Sorry, was away on a different project, now I'm back...
I tried to use the classes property to bring back the highway-key. For this I set up my profile like this:

`classes = Sequence {
        'highway', 'toll', 'motorway', 'ferry', 'restricted', 'tunnel'
    },`

But I might have a general misunderstanding on how this is supposed to work. I understood it should bring me back these configured classes, if intersection(0) is of that type.
I do get back for example 'motorway' on motorway-intersections, but how can I get the highway-key?

Thanks a lot for your help!

**Comment by linuzer** [comment]
For the protocol:
I found an easier solution: In Germany, the new truck-toll system applies to all "Autobahnen" (=highway:motorway) and "Bundesstrassen" (=highway:primary). Since almost all of these mayor roads do have a correct Ref-key set ("A 44", "B 9"), I was able to loop through the navigation steps. Whenever a ref-key was set and it starts with "A" or "B" I take the distance of that step and add it to the toll-km. The result is the very accurate total toll-km of that route. Voilà.

The suggested toll-key is almost useless in Germany, since it is almost never set (or has a different meaning)!
